<a href="https://colab.research.google.com/github/TaranSchlich/Real-Estate-Valuation-UW-Madison-MSDIA/blob/main/GB881_Assignment_5_Schlichtmann_T.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Real Estate Valuation

By: Taran Schlichtmann

### Objective

The objective of this notebook is to explore the correlation between various variables and real estate value by using a real estate dataset. By performing data analysis, computing correlations, and visualizing relationships of different variables to determine their relationship with home value.

### Import Libraries

In [ ]:
# Libraries used for analysis, data manipulation, and visualization

import pandas as pd                 # importing dataset and data manipulation
import seaborn as sns               # data visualization tools

### Dataframe Creation

In [ ]:
# Loading 'Real Estate CSV' file into dataframe

url_Real_Estate = 'http://bit.ly/real-estate-data'

Dataframe_Real_Estate = pd.read_csv(url_Real_Estate)

### Dimensions of Dataset

In [ ]:
# Determining rows and columns of dataset

Dataframe_Real_Estate.shape

(414, 8)

### Columns and Datatypes

In [ ]:
# Inspecting column names, datatypes, and completeness with .info()

Dataframe_Real_Estate.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 414 entries, 0 to 413
Data columns (total 8 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   No                                      414 non-null    int64  
 1   X1 transaction date                     414 non-null    float64
 2   X2 house age                            414 non-null    float64
 3   X3 distance to the nearest MRT station  414 non-null    float64
 4   X4 number of convenience stores         414 non-null    int64  
 5   X5 latitude                             414 non-null    float64
 6   X6 longitude                            414 non-null    float64
 7   Y house price of unit area              414 non-null    float64
dtypes: float64(6), int64(2)
memory usage: 26.0 KB


### Dataset Overview, Shape, and Integrity
Dataset contains 414 records of real estate transactions, including house age, proximity to MRT stations, and unit price. All of the columns have 414 non-null entries with no missing data. Most fields are 'float64', with a few 'int64', making them suitable for correlation and regression analysis.

### Adjusting fields

In [ ]:
# improving readability of field names

Dataframe_Real_Estate = Dataframe_Real_Estate.rename(columns={'No' : 'sale_number',
                                                              'X1 transaction date' : 'sale_date',
                                                              'X2 house age' : 'house_age',
                                                              'X3 distance to the nearest MRT station' : 'distance_to_station',
                                                              'X4 number of convenience stores' : 'num_convenience_stores',
                                                              'X5 latitude' : 'latitude',
                                                              'X6 longitude' : 'longitude',
                                                              'Y house price of unit area' : 'home_price_per_unit'})

Improved column names to make the dataset more intuitive and easier to work with during analysis.

### Preview of Data

In [ ]:
# Previewing the first 7 records in the dataset

Dataframe_Real_Estate.head(7)

Previewing the data to verify that columns were renamed correctly and see formatting.

### Descriptive Statistics

In [ ]:
# Viewing statistics of the dataset

Dataframe_Real_Estate.describe()

This output reveals the tendency and spread of each variable. The average house age is around 17.7 years, and the average unit price is approximately 37.98.

### Histogram

In [ ]:
# visualizing the distribution of data in the housing price column

sns.displot(data=Dataframe_Real_Estate, x='home_price_per_unit')

Note that the price tends to center around around 40.

### Scatterplot

In [ ]:
# visualizing a scatterplot for the relationship between the age and price of a home

sns.scatterplot(data=Dataframe_Real_Estate, x='house_age', y='home_price_per_unit')

The scatterplot appears to show a weak negative relationship as older homes tend to be slightly less expensive but the trend is not very strong.

### Correlation Matrix

In [ ]:
# Generating a correlation matrix with correlation coefficients for all columns except longitude, latitude, and transaction date

corr = Dataframe_Real_Estate.iloc[:, [0, 2, 3, 4, 7]].corr()
print(corr)

Distance to MRT station has a strong negative correlation with price -0.67, suggesting proximity boosts value. Number of convenience stores shows a moderate positive correlation 0.57, indicating neighborhood amenities matter. House age has a weak negative correlation -0.21, consistent with earlier scatterplot findings.

### Heatmap

In [ ]:
# Visualizing a heatmap that includes the correlation coefficients for all columns except longitude, latitude, and transaction date

sns.heatmap(corr, annot=True)               # annot=True to include correlation coefficients

The heatmap confirms that distance to MRT station is the most influential variable negatively affecting price, while convenience store count positively correlates with value.

# Summary

This analysis highlights key drivers of real estate value in the dataset. Proximity to MRT stations significantly increases home value, while neighborhood amenities such as convenience stores also contribute positively. House age shows a minor negative impact on price.